File: 0566-reshape-the-matrix/0566-reshape-the-matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        int ans[][]=new int[r][c];
        int row=0;
        int col=0;
        if(mat[0].length*mat.length!=r*c){
            return mat;
        }
        for(int i=0;i<mat.length;i++){
            for(int j=0;j<mat[0].length;j++){
                ans[row][col]=mat[i][j];
                col++;
                if(col==c){
                    row++;
                    col=0;
                }
            }
        }
        return ans;
    }
}

File: 0566-reshape-the-matrix/0566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        ans=[[0]*c for i in range(r)]
        if len(mat)*len(mat[0])!=r*c:
            return mat
        row=0
        col=0
        for i in range(len(mat)):
            for j in range(len(mat[0])):
                ans[row][col]=mat[i][j]
                col+=1
                if(col==c):
                    col=0
                    row+=1

        return ans